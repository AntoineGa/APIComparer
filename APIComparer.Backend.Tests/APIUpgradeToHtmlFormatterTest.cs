namespace APIComparer.Tests.VersionComparisons
{
    using System.IO;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using APIComparer.Backend;
    using APIComparer.Backend.Reporting;
    using APIComparer.VersionComparisons;
    using NUnit.Framework;

    [TestFixture]
    [UseReporter(typeof(DiffReporter),typeof(ClipboardReporter))]
    public class APIUpgradeToHtmlFormatterTest
    {
        [Test]
        public void TestComplianceNewtonsoftJson()
        {
            Approvals.VerifyHtml(HtmlDiff("newtonsoft.json", "5.0.8", "6.0.8"));
        }

        [Test]
        public void TestComplianceNServiceBus()
        {
            Approvals.VerifyHtml(HtmlDiff("nservicebus", "4.0.0", "5.0.0"));
        }

        [Test]
        public void TestComplianceAzureSelfDestruct()
        {
            Approvals.VerifyHtml(HtmlDiff("Two10.Azure.SelfDestruct", "1.0.0", "1.0.5"));
        }

        [Test]
        public void TestComplianceLibLog()
        {
            Approvals.VerifyHtml(HtmlDiff("LibLog", "3.0.0", "4.1.1"));
        }

        [Test]
        public void AppccelerateEventBrokerNoChanges()
        {
            Approvals.VerifyHtml(HtmlDiff("Appccelerate.EventBroker", "3.1.0", "3.15.0"));
        }

        private string HtmlDiff(string packageName, string fromVersion, string untilVersion)
        {
            using (var writer = new StringWriter())
            {
                var formatter = new APIUpgradeToHtmlFormatter();

                var packageDescription = new PackageDescription
                {
                    PackageId = packageName,
                    Versions = new VersionPair(fromVersion, untilVersion)
                };

                var compareSetCreator = new CompareSetCreator();
                var sets = compareSetCreator.Create(packageDescription);
                var compareSetDiffer = new CompareSetDiffer();
                var diff = compareSetDiffer.Diff(sets);

                formatter.Render(writer, packageDescription, diff);

                return writer.ToString();
            }
        }
    }
}
