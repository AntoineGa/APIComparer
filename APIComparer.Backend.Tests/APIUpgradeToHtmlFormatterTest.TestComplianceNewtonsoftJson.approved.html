
<h1>.NETPortable,Version=v0.0,Profile=net45+wp80+win8+wpa81+aspnetcore50 (Compared To .NETPortable,Version=v0.0,Profile=net45+wp80+win8)</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.JsonConvert</h3>













  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    
<h1>.NETPortable,Version=v0.0,Profile=net45+wp80+win8</h1>
    <p>No longer supported.</p>

<h1>.NETPortable,Version=v0.0,Profile=net40+sl5+wp80+win8+wpa81 (Compared To .NETPortable,Version=v0.0,Profile=net45+wp80+win8)</h1>
        
<h2>The following public types have been removed</h2>
<ul>
  <li><code>Newtonsoft.Json.Converters.ExpandoObjectConverter</code></li>
  <li><code>Newtonsoft.Json.Converters.XmlNodeConverter</code></li>
  <li><code>Newtonsoft.Json.Serialization.JsonDynamicContract</code></li>
</ul>
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.JsonConvert</h3>









<h4>Methods Removed</h4>
<ul>
  <li><code>System.Threading.Tasks.Task&lt;T&gt; DeserializeObjectAsync&lt;T&gt;(string)</code></li>
  <li><code>System.Threading.Tasks.Task&lt;T&gt; DeserializeObjectAsync&lt;T&gt;(string, Newtonsoft.Json.JsonSerializerSettings)</code></li>
  <li><code>System.Threading.Tasks.Task&lt;object&gt; DeserializeObjectAsync(string)</code></li>
  <li><code>System.Threading.Tasks.Task&lt;object&gt; DeserializeObjectAsync(string, System.Type, Newtonsoft.Json.JsonSerializerSettings)</code></li>
  <li><code>System.Xml.Linq.XDocument DeserializeXNode(string)</code></li>
  <li><code>System.Xml.Linq.XDocument DeserializeXNode(string, string)</code></li>
  <li><code>System.Xml.Linq.XDocument DeserializeXNode(string, string, bool)</code></li>
  <li><code>System.Threading.Tasks.Task PopulateObjectAsync(string, object, Newtonsoft.Json.JsonSerializerSettings)</code></li>
  <li><code>System.Threading.Tasks.Task&lt;string&gt; SerializeObjectAsync(object)</code></li>
  <li><code>System.Threading.Tasks.Task&lt;string&gt; SerializeObjectAsync(object, Newtonsoft.Json.Formatting)</code></li>
  <li><code>System.Threading.Tasks.Task&lt;string&gt; SerializeObjectAsync(object, Newtonsoft.Json.Formatting, Newtonsoft.Json.JsonSerializerSettings)</code></li>
  <li><code>string SerializeXNode(System.Xml.Linq.XObject)</code></li>
  <li><code>string SerializeXNode(System.Xml.Linq.XObject, Newtonsoft.Json.Formatting)</code></li>
  <li><code>string SerializeXNode(System.Xml.Linq.XObject, Newtonsoft.Json.Formatting, bool)</code></li>
</ul>




  <h3>Newtonsoft.Json.Linq.JContainer</h3>









<h4>Methods Removed</h4>
<ul>
  <li><code>void add_CollectionChanged(System.Collections.Specialized.NotifyCollectionChangedEventHandler)</code></li>
  <li><code>void remove_CollectionChanged(System.Collections.Specialized.NotifyCollectionChangedEventHandler)</code></li>
</ul>




  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    
<h1>.NETPortable,Version=v0.0,Profile=net40+sl4+wp7+win8</h1>
    <p>No longer supported.</p>

<h1>.NETFramework,Version=v4.5</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.JsonConvert</h3>













  <h3>Newtonsoft.Json.Linq.JPropertyDescriptor</h3>









<h4>Methods Removed</h4>
<ul>
  <li><code>void .ctor(string, Type)</code></li>
</ul>




  <h3>Newtonsoft.Json.Serialization.DefaultContractResolver</h3>













  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    
<h1>.NETFramework,Version=v4.0</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.JsonConvert</h3>













  <h3>Newtonsoft.Json.Linq.JPropertyDescriptor</h3>









<h4>Methods Removed</h4>
<ul>
  <li><code>void .ctor(string, Type)</code></li>
</ul>




  <h3>Newtonsoft.Json.Serialization.DefaultContractResolver</h3>













  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    
<h1>.NETFramework,Version=v3.5</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.Linq.JPropertyDescriptor</h3>









<h4>Methods Removed</h4>
<ul>
  <li><code>void .ctor(string, Type)</code></li>
</ul>




  <h3>Newtonsoft.Json.Serialization.DefaultContractResolver</h3>













  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    
<h1>.NETFramework,Version=v2.0</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.Linq.JPropertyDescriptor</h3>









<h4>Methods Removed</h4>
<ul>
  <li><code>void .ctor(string, Type)</code></li>
</ul>




  <h3>Newtonsoft.Json.Serialization.DefaultContractResolver</h3>













  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    
<h1>.NETCore,Version=v4.5</h1>
        
        
        
<h2>The following types have differences.</h2>

  <h3>Newtonsoft.Json.JsonConvert</h3>













  <h3>Newtonsoft.Json.Serialization.JsonObjectContract</h3>













    